pkg/auth/metadata.go

```diff
@@ -206,13 +214,36 @@ func (m *KVMetadataManager) IsInitialized(ctx context.Context) (bool, error) {
 	return !setupTimestamp.IsZero(), nil
 }
 
+// For testing purposes
+var StatFunc = os.Stat
+
+func getInstrumentation() string {
+	_, k8s := os.LookupEnv("KUBERNETES_SERVICE_HOST")
+	lakefsAccessKeyID := os.Getenv("LAKEFS_ACCESS_KEY_ID")
+	quickstart, _ := strconv.ParseBool(os.Getenv("QUICKSTART"))
+	_, statErr := StatFunc("/.dockerenv")
```

Comment:
1. separate keys for docker / k8s / sample|quickstart
2. stats - global string point to dockerenv filename instead of override stat func